### PR TITLE
added code for handling cc and bcc mail

### PR DESCRIPTION
**Pull request by ravisaurav-tarento**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: notification/notification-sdk/src/main/java/org/sunbird/notification/email/service/impl/SmtpEMailServiceImpl.java [hunk 1]
```diff
@@ -28,19 +28,24 @@ public boolean sendEmail(EmailRequest emailReq) {
       logger.info("Email request is null or empty:");
       return false;
       // either email object has bcc or to list size more than 1 then pass it as bcc.
-    } else if (CollectionUtils.isNotEmpty(emailReq.getBcc()) || emailReq.getTo().size() > 1) {
+    } else if (CollectionUtils.isNotEmpty(emailReq.getBcc()) && !emailReq.getTo().isEmpty()) {
```
Comment (karthik-tarento) [inline notification/notification-sdk/src/main/java/org/sunbird/notification/email/service/impl/SmtpEMailServiceImpl.java]: This is not proper approach... 
the existing code allowed to send email when bcc is not empty OR to address is not empty... 
but latest code allows only when the bbc is not empty and to address is not empty... -- what will happen to existing implementation -- does all the cases, we have bcc and to available ??

File: notification/notification-sdk/src/main/java/org/sunbird/notification/email/Email.java [hunk 2]
```diff
@@ -194,11 +194,12 @@ public void sendAttachment(
    * @param bccList recipient bcc list
    * @return boolean
    */
-  public boolean sendEmail(String fromEmail, String subject, String body, List<String> bccList) {
+  public boolean sendEmail(String fromEmail, String subject, String body,  List<String> toList, List<String> bccList) {
     boolean sentStatus = true;
     try {
       Session session = getSession();
       MimeMessage message = new MimeMessage(session);
+      addRecipient(message, RecipientType.TO, toList);
```
Comment (karthik-tarento) [inline notification/notification-sdk/src/main/java/org/sunbird/notification/email/Email.java]: Why this is added as "TO" -- suppose to be "CC" ?

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: notification/notification-sdk/src/main/java/org/sunbird/notification/email/service/impl/SmtpEMailServiceImpl.java (hunk 1 above)
Comment (ravisaurav-tarento) [reply notification/notification-sdk/src/main/java/org/sunbird/notification/email/service/impl/SmtpEMailServiceImpl.java]: This code has been removed Karthik

File: notification/notification-sdk/src/main/java/org/sunbird/notification/email/Email.java (hunk 2 above)
Comment (ravisaurav-tarento) [reply notification/notification-sdk/src/main/java/org/sunbird/notification/email/Email.java]: This code has been modified and a new approach has been implemented.

**Pull request merged**